src/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.CoreCLR.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+#if BIT64
+using nint = System.Int64;
+#else
+using nint = System.Int32;
+#endif
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    public partial struct GCHandle
+    {
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern IntPtr InternalAlloc(object value, GCHandleType type);
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern void InternalFree(IntPtr handle);
+
+#if DEBUG
+        // The runtime performs additional checks in debug builds
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern object InternalGet(IntPtr handle);
+#else
+        internal static unsafe object InternalGet(IntPtr handle) =>
```

Comment:
Could this go to shared partition?

Reply:
corert doesn't use this, and coreclr doesn't use it in debug. It's one line; cleanest thing would seem to be to just dup that line in mono's partial file.

Reply:
What is the extra logic for the debug build? I did a quick search and found only some AppDomain check which I am not sure is still relevant

Reply:
It ends up doing a VALIDATEOBJECTREF to validate that the pointer actually is an object rather than blindly trusting it. Without buy-in from Jan, I wouldn't want to remove it.

But even if we did, the implementation is still one line, differs from corert, and is inherently tied to the runtime's GC's representation of handles. Would this line even work on mono?

Reply:
Right, `VALIDATEOBJECTREF` is a check that is present in debug build of the runtime only. We like to have these checks sprinkled around for stress. These object validation checks help us catch stress bugs closer to the root cause.